src/validation.js

```diff
@@ -0,0 +1,45 @@
+import CONSTANT from "./constant.js";
+
+const validation = {
+  numberValidation: (value) => {
+    if (isNaN(value)) return true;
+
+    const isRangeCheck = value >= 1 && value <= 50;
+    return !isRangeCheck;
+  },
+  sameValidation: (value, temp) => {
+    const isSameValue = temp.some((tempValue) => tempValue === value);
+    return isSameValue;
+  },
+
+  validateInput: (input, tempValue) => {
+    const regex = new RegExp(`^${tempValue}$`, "i");
+    return regex.test(input);
+  },
+
+  isSameValue: (value, temp) => {
+    return value === temp;
+  },
+
+  main: (type, temp, tempValue) => {
```

Comment:
P4;

개인적으로 params가 3개 이상(상황에따라선 2개 이상)일 경우에는 object를 권장드려요.

object로 넘길 때 가독성이 좋아질 뿐 아니라 매개변수 순서를 기억 할 필요가 없어지기도 하고, optional 매개변수를 만들때에도 훨씬 쉬워요. 추가로 새로운 매개변수를 추가, 삭제할 때 유지보수 측면에서 훨씬 관리하기 편해지는 잇점도 있어요.

```suggestion
  main: ({ type, temp, tempValue }) => {
```